oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -47,29 +53,79 @@ public static List<Rule> loadRules(final String path) throws ModuleStartExceptio
+            rules = ResourceUtils.list(path);
         } catch (FileNotFoundException e) {
             throw new ModuleStartException("Load fetcher rules failed", e);
         }
-        return Arrays.stream(rules)
-            .filter(File::isFile)
-            .map(f -> {
-                try (Reader r = new FileReader(f)) {
-                    String fileName = f.getName();
-                    int dotIndex = fileName.lastIndexOf('.');
-                    fileName = (dotIndex == -1) ? fileName : fileName.substring(0, dotIndex);
-                    if (!enabledRules.contains(fileName)) {
-                        return null;
+        Map<String, Boolean> formedEnabledRules = enabledRules
+                .stream()
+                .map(rule -> {
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (rule.endsWith(".yaml")) {
+                        return rule;
+                    } else if (rule.endsWith(".yml")) {
+                        return rule.substring(0, rule.length() - 2) + "aml";
+                    } else {
+                        return rule + ".yaml";
                     }
-                    Rule rule = new Yaml().loadAs(r, Rule.class);
-                    rule.setName(fileName);
-                    return rule;
-                } catch (IOException e) {
-                    LOG.debug("Reading file {} failed", f, e);
+                })
+                .collect(Collectors.toMap(rule -> rule, $ -> false));
+
+        List<Rule> result = Arrays.stream(rules)
+                .flatMap(f -> {
+                    if (f.isDirectory()) {
+                        return Arrays.stream(Objects.requireNonNull(f.listFiles()))
+                                .filter(File::isFile)
+                                .map(file -> getRulesFromFile(formedEnabledRules, f, file));
+                    } else {
+                        return Stream.of(getRulesFromFile(formedEnabledRules, null, f));
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        if (formedEnabledRules.containsValue(false)) {
+            throw new UnexpectedException("Some configuration files of enabled rules are not found, enabled rules: " + formedEnabledRules.keySet());
+        }
+        return result;
+    }
+
+    private static Rule getRulesFromFile(Map<String, Boolean> formedEnabledRules, File directory, File file) {
+        try (Reader r = new FileReader(file)) {
+            String fileName = file.getName();
+            int dotIndex = fileName.lastIndexOf('.');
+            if (fileName.endsWith(".yml")) {
+                fileName = fileName.substring(0, fileName.length() - 2) + "aml";
+            }
+            if (dotIndex == -1 || !"yaml".equals(fileName.substring(dotIndex + 1))) {
+                return null;
+            }
```

Comment:
Why do we change yml to yaml? These lines seems meaning we support file with `.yml` and `.yaml` suffixes? Why don't we put the codes in that way?

Reply:
![image](https://user-images.githubusercontent.com/74546965/197661583-b2f0e2ab-62d8-46c2-9ecd-bb52e6bc3105.png)
For this, we support .yml.
And this will cause a problem : enabledRule: mysql means mysql.yml or mysql.yaml ? So I just change yml to yaml to support both situation. The only problem with such code is that if there are both mysql.yml and mysql.yaml, there is no guarantee which one will be loaded.

Reply:
I know we should support both. I mean the way you put in codes is strange. Taking two suffixes should do directly, rather than a rename. In your way, we could(maybe not now) output the wrong file causing confusing.

Reply:
I understand, I will try it later.